### DISCUSSION TITLE
Custom verification for multiple domains - redirect to correct domain on verify

### DISCUSSION THREAD

**Discussion by sfdc-ben** [pocketbase/pocketbase#2260] [Q&A] [answered]
I have set up a custom verification using the confirmVerification method, however I am having trouble accepting context for the redirect. I can only pass the token on the Action URL in the verification email, and the confirmVerification method returns null on success.

Is there a way to know the email of the user being verified? If yes, I can get the source domain and redirect appropriately (I have it logged as an attribute on the user record on create through a registration page).

_Use case: multiple front-end sites utilizing the same auth database, with registration pages on each._

**Comment by ganigeorgiev** [comment, marked as answer]
> Is there a way to know the email of the user being verified?

You can extract the auth record id and email from the JWT token payload, something like:

```js
const jwt = "...";
const payload = JSON.parse(atob(jwt.split(".")[1])) // base64 decode the middle part and then unmarshalize
```
